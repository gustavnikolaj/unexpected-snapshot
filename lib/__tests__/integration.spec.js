const jestExpect = global.expect;
const expect = require('unexpected')
    .clone()
    .use(require('../unexpectedSnapshot'));

it('foo', () => {
    expect('foo', 'to match snapshot');
});

it('to throw error matching snapshot', () => {
    expect(
        () => expect('foobar', 'to equal', 'foo'),
        'to throw error matching snapshot'
    );
});

it('<object> to match snapshot simple', () => {
    expect({ foo: 'barfoo' }, 'to match snapshot');
});

it('<object> to match snapshot with circular reference', () => {
    const obj = { foo: { bar: 'baz' } };
    obj.obj = obj;

    expect(obj, 'to match snapshot');
});

it('<object> to match snapshot with functions', () => {
    const obj = {
        a: 'wholeLotOfFunctions',
        af: foo => foo + 'bar',
        afb: foo => {
            return foo + 'bar';
        },
        anf: function (foo) { return foo + 'bar'; }
    };

    expect(obj, 'to match snapshot');
});

it('to throw error matching snapshot native', () => {
    jestExpect(() =>
        expect('foobar', 'to equal', 'fooba')
    ).toThrowErrorMatchingSnapshot();
});
