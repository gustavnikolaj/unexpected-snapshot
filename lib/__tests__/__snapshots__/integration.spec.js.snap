// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<object> to match snapshot simple 1`] = `"{ foo: 'barfoo' }"`;

exports[`<object> to match snapshot with circular reference 1`] = `"{ foo: { bar: 'baz' }, obj: [Circular] }"`;

exports[`<object> to match snapshot with functions 1`] = `
"{ a: 'wholeLotOfFunctions',
  af: [Function: af],
  afb: [Function: afb],
  anf: [Function: anf] }"
`;

exports[`foo 1`] = `"foo"`;

exports[`to throw error matching snapshot 1`] = `
"expected 'foobar' to equal 'foo'

-foobar
+foo"
`;

exports[`to throw error matching snapshot native 1`] = `
"
[31m[1mexpected[22m[39m [36m'foobar'[39m [31m[1mto equal[22m[39m [36m'fooba'[39m

[41m[30mfoobar[39m[49m
[42m[30mfooba[39m[49m
"
`;
